Detectors/Base/src/Propagator.cxx

```diff
@@ -47,6 +49,52 @@ Propagator::Propagator()
   mField->GetBz(xyz, mBz);
 }
 
+//____________________________________________________________
+int Propagator::initFieldFromGRP(const std::string grpFileName, std::string grpName, bool verbose)
+{
+  /// load grp and init magnetic field
+  if (verbose) {
+    LOG(INFO) << "Loading field from GRP of " << grpFileName;
+  }
+  const auto grp = o2::parameters::GRPObject::loadFrom(grpFileName, grpName);
+  if (!grp) {
+    return -1;
+  }
+  if (verbose) {
+    grp->print();
+  }
+
+  return initFieldFromGRP(grp);
+}
+
+//____________________________________________________________
+int Propagator::initFieldFromGRP(const o2::parameters::GRPObject* grp, bool verbose)
+{
+  /// init mag field from GRP data and attach it to TGeoGlobalMagField
+
+  if (TGeoGlobalMagField::Instance()->IsLocked()) {
+    if (TGeoGlobalMagField::Instance()->GetField()->TestBit(o2::field::MagneticField::kOverrideGRP)) {
+      LOG(WARNING) << "ExpertMode!!! GRP information will be ignored";
+      LOG(WARNING) << "ExpertMode!!! Running with the externally locked B field";
+      return 0;
+    } else {
+      LOG(INFO) << "Destroying existing B field instance";
+      delete TGeoGlobalMagField::Instance();
+    }
+  }
+  auto fld = o2::field::MagneticField::createFieldMap(grp->getL3Current(), grp->getDipoleCurrent());
+  TGeoGlobalMagField::Instance()->SetField(fld);
+  TGeoGlobalMagField::Instance()->Lock();
+  if (verbose) {
+    LOG(INFO) << "Running with the B field constructed out of GRP";
+    LOG(INFO) << "Access field via TGeoGlobalMagField::Instance()->Field(xyz,bxyz) or via";
+    LOG(INFO) << "auto o2field = static_cast<o2::field::MagneticField*>( TGeoGlobalMagField::Instance()->GetField() )";
+  }
+  return 0;
+}
+
+#endif
```

Comment:
???

Reply:
I moved it to another place in the file to have the functions hidden by `#ifndef` in one place.